system-apps/desktop-documentation-viewer/webClient/src/App.tsx

```diff
@@ -0,0 +1,168 @@
+import { withTranslation } from 'react-i18next';
+
+class App extends React.Component<any, any> {
+  private log: ZLUX.ComponentLogger;
+  t: any;
+  constructor(props){
+    super(props);
+    this.log = this.props.resources.logger;
+    let metadata = this.props.resources.launchMetadata;
+    if (metadata != null && metadata.data != null && metadata.data.type != null) {
+      this.handleLaunchOrMessageObject(metadata.data);
+    } else {
+      this.state = this.getDefaultState();
+    }
+
+  };
+
+  private getDefaultState() {
+    return {
+      actionType: "Launch",
+      appTarget: "PluginCreate",
+      docsList: "",
+      pluginIdentifier: "",
+      allPlugins: ZoweZLUX.pluginManager.loadPlugins("application", false).__zone_symbol__value,
+      destination: ZoweZLUX.uriBroker.pluginRESTUri(this.props.resources.pluginDefinition.getBasePlugin(), 'plugindetail', "")
+    };
+  }
+
+  updateOrInitState(stateUpdate: any): void {
+    if (!this.state) {
+      this.state = Object.assign(this.getDefaultState(), stateUpdate);
+    }
+    else {
+      this.setState(stateUpdate);
+    }
+  }
+
+  handleLaunchOrMessageObject(data: any) {
+    switch (data.type) {
+    case 'setAppRequest':
+      let actionType = data.actionType;
+      let msg:string;
+      if (actionType == 'Launch' || actionType == 'Message') {
+        let mode = data.targetMode;
+        if (mode == 'PluginCreate' || mode == 'PluginFindAnyOrCreate') {
+          this.updateOrInitState({actionType: actionType,
+                                  appTarget: mode,
+                                  appId: data.targetAppId,
+                                  parameters: data.requestText});
+        } else {
+          msg = `Invalid target mode given (${mode})`;
+          this.log.warn(msg);
+          this.updateOrInitState({status: msg});
+        }
+      } else {
+        msg = `Invalid action type given (${actionType})`;
+        this.log.warn(msg);
+        this.updateOrInitState({status: msg});
+      }
+      break;
+    default:
+      this.log.warn(`Unknown command (${data.type}) given in launch metadata.`);
+    }
+  }
+
+  zluxOnMessage(eventContext: any): Promise<any> {
+    return new Promise((resolve,reject)=> {
+      if (eventContext != null && eventContext.data != null && eventContext.data.type != null) {
+        resolve(this.handleLaunchOrMessageObject(eventContext.data));
+      } else {
+        let msg = 'Event context missing or malformed';
+        this.log.warn('onMessage '+msg);
+        return reject(msg);
+      }
+    });
+  }
+  
+  provideZLUXDispatcherCallbacks(): ZLUX.ApplicationCallbacks {
+    return {
+      onMessage: (eventContext: any): Promise<any> => {
+        return this.zluxOnMessage(eventContext);
+      }      
+    }
+  }  
+
+  getPluginsFromServer() {
+    let allPlugins = ZoweZLUX.pluginManager.loadPlugins("application", false);
+    this.updateOrInitState({allPlugins: allPlugins.__zone_symbol__value})
```

Comment:
unsure if __zone_symbol__value will be good to use long-term, because I think this is an angular object so if we upgrade from angular 6 to 8, 9, etc the variable name could change or disappear. I wonder if there's a better way to get this data?